### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by defaultuser001**
Check it out, please

regards!

### CODE REVIEW

File: winappdbg/debug.py [hunk 1]
```diff
@@ -505,7 +507,8 @@ def execl(self, lpCmdLine, **kwargs):
             # Warn when mixing 32 and 64 bits.
             # This also allows the user to stop attaching altogether,
             # depending on how the warnings are configured.
-            if System.bits != aProcess.get_bits():
+            BITS_WARNING_FLAG = False
```
Comment (MarioVilas) [inline winappdbg/debug.py:510]: I'm curious about this, why remove the warning?

File: winappdbg/debug.py [hunk 2]
```diff
@@ -1079,16 +1082,21 @@ def next(self):
             event handler raises an exception nobody catches.
         """
         try:
-            event = self.wait()  # NOQA
+            try:
+                event = self.wait(time_limit_in_seconds * 1000)  # NOQA
+            except WindowsError:
+                if time_limit_in_seconds == 0:
+                    pass
```
Comment (MarioVilas) [inline winappdbg/debug.py:1089]: Wouldn't this make it ignore all exceptions instead of raising them again?

File: winappdbg/win32/kernel32.py [hunk 3]
```diff
@@ -3707,8 +3707,8 @@ def ReadProcessMemory(hProcess, lpBaseAddress, nSize):
     lpBuffer            = ctypes.create_string_buffer(b'', nSize)
     lpNumberOfBytesRead = SIZE_T(0)
     success = _ReadProcessMemory(hProcess, lpBaseAddress, lpBuffer, nSize, byref(lpNumberOfBytesRead))
-    if not success and GetLastError() != ERROR_PARTIAL_COPY:
-        raise ctypes.WinError()
+    #if not success and GetLastError() != ERROR_PARTIAL_COPY:
```
Comment (MarioVilas) [inline winappdbg/win32/kernel32.py:3710]: This was kinda important to keep the memory dumps from failing on corrupted processes, why was it removed?

File: winappdbg/event.py [hunk 4]
```diff
@@ -1398,7 +1398,7 @@ def __init__(self, myArgument):
         for lib, hooks in self.apiHooks.items():
             hook_objs = []
             for proc, args in hooks:
-                if type(args) in (int, long):
+                if type(args) in (int, int):
```
Comment (MarioVilas) [inline winappdbg/event.py:1401]: Seems redundant :D

File: winappdbg/event.py [hunk 5]
```diff
@@ -1849,7 +1849,7 @@ def dispatch(self, event):
                 except Exception as e:
                     msg = ("Event handler pre-callback %r"
                            " raised an exception: %s")
-                    msg = msg % (self.__eventHandler, traceback.format_exc(e))
+                    msg = msg % (self.__eventHandler, traceback.format_exc())
```
Comment (MarioVilas) [inline winappdbg/event.py:1852]: Probably should use format_exception() instead

File: tools/run_01.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
+import winappdbg
```
Comment (MarioVilas) [inline tools/run_01.py:1]: Feels like this is a unit test more than a tool...

File: tools/service.py [hunk 7]
```diff
@@ -198,7 +198,7 @@ def show(search = None, wide = True):
     # Sort the rows.
     data = sorted(data)
 
-    # Build the table and print it.
+    # Build the table and print(it.)
```
Comment (MarioVilas) [inline tools/service.py:201]: :D

File: winappdbg/process.py [hunk 8]
```diff
@@ -3963,9 +3982,18 @@ def get_windows(self):
         @return: Returns a list of windows
             handled by all processes in this snapshot.
         """
-        window_list = list()
+        '''window_list = dict()
         for process in self.iter_processes():
             window_list.extend( process.get_windows() )
+        return window_list'''
+
+        hCurWnd = None
```
Comment (MarioVilas) [inline winappdbg/process.py:3997]: Nice one! :)

File: winappdbg/sql.py [hunk 9]
```diff
@@ -180,7 +180,7 @@ class BaseDAO (object):
         L{_session} instance variable.
 
     @type _echo: bool
-    @cvar _echo: Set to C{True} to print all SQL queries to standard output.
+    @cvar _echo: Set to C{True} to print(all SQL queries to standard output.)
```
Comment (MarioVilas) [inline winappdbg/sql.py:183]: :D

File: winappdbg/textio.py [hunk 10]
```diff
@@ -1788,7 +1788,7 @@ class Logger(object):
     @ivar logfile: Append messages to this text file.
 
     @type verbose: bool
-    @ivar verbose: C{True} to print messages to standard output.
+    @ivar verbose: C{True} to print(messages to standard output.)
```
Comment (MarioVilas) [inline winappdbg/textio.py:1788]: Seems to be a pattern with py2to3...

File: winappdbg/textio.py [hunk 11]
```diff
@@ -1801,7 +1801,7 @@ def __init__(self, logfile = None, verbose = True):
         @param logfile: Append messages to this text file.
 
         @type  verbose: bool
-        @param verbose: C{True} to print messages to standard output.
+        @param verbose: C{True} to print(messages to standard output.)
```
Comment (MarioVilas) [inline winappdbg/textio.py:1801]: Another one

File: winappdbg/textio.py [hunk 12]
```diff
@@ -1875,7 +1876,7 @@ def log_exc(self):
 
     def is_enabled(self):
         """
-        Determines if the logger will actually print anything when the log_*
+        Determines if the logger will actually print(anything when the log_*)
```
Comment (MarioVilas) [inline winappdbg/textio.py:1876]: :D

File: winappdbg/util.py [hunk 13]
```diff
@@ -765,7 +768,10 @@ class DebugRegister (StaticClass):
     )
 
     # Dr7 &= disableMask[register]
-    disableMask = tuple( [registerMask ^ x for x in enableMask] )
+    disableMask = []
+    for x in enableMask:
```
Comment (MarioVilas) [inline winappdbg/util.py:772]: Wouldn't this be slower? (Not sure how optimizations work in py3)

File: winappdbg/win32/kernel32.py [hunk 14]
```diff
@@ -587,7 +587,7 @@ class Handle (object):
     """
 
     # XXX DEBUG
-    # When this private flag is True each Handle will print a message to
+    # When this private flag is True each Handle will print(a message to)
```
Comment (MarioVilas) [inline winappdbg/win32/kernel32.py:590]: :D

File: winappdbg/win32/peb_teb.py [hunk 15]
```diff
@@ -2217,7 +2217,7 @@ class _TEB_NT(Structure):
 #    +0xf2c Instrumentation  : [16] Ptr32 Void
 #    +0xf6c WinSockData      : Ptr32 Void
 #    +0xf70 GdiBatchCount    : Uint4B
-#    +0xf74 InDbgPrint       : UChar
+#    +0xf74 InDbgprint(      : UChar)
```
Comment (MarioVilas) [inline winappdbg/win32/peb_teb.py:2220]: I'm starting to get annoyed by this tool... I mean really

File: winappdbg/win32/user32.py [hunk 16]
```diff
@@ -71,7 +71,7 @@ class __WindowEnumerator (object):
     def __init__(self):
         self.hwnd = list()
     def __call__(self, hwnd, lParam):
-##        print hwnd  # XXX DEBUG
+##        print(hwnd  # XXX DEBUG)
```
Comment (MarioVilas) [inline winappdbg/win32/user32.py:74]: OMG so broken hahaha

I get it, automated porting tools are hard, but why try to parse what's inside comments anyway?

**Comment by MarioVilas** [comment]
This is so cool! I left a bunch of comments in the code, let me know what you think

**Comment by MarioVilas** [comment]
Oh, just saw the conflict in pfind.py. It's easy to resolve though, it's just a couple missing if statements in the py3 version.

File: winappdbg/debug.py (hunk 1 above)
Comment (defaultuser001) [reply winappdbg/debug.py:510]: Sorry about that, it was for my project and i forgot to remove this flag

File: winappdbg/debug.py (hunk 2 above)
Comment (defaultuser001) [reply winappdbg/debug.py:1089]: Yes, so it seems, noob mistake, sorry

Comment (MarioVilas) [reply winappdbg/debug.py:1089]: No prob! That's what peer review is for! :)

**Pull request closed without merging**